docs/source/usage.rst

```diff
@@ -534,23 +534,36 @@ namespace, and to use interpolations in the name itself. The following example d
     4
 
 
-By default a custom resolver's output is cached, so that when it is called with the same
-inputs we always return the same value. This behavior may be disabled by setting ``use_cache=False``:
+By default a custom resolver is called on every access, but it is possible to cache its output
+by registering it with ``use_cache=True``.
+This may be useful either for performance reasons or to ensure the same value is always returned.
+Note that the cache is based on the string literals representing the resolver's inputs, and not
+the inputs themselves:
 
 .. doctest::
 
     >>> import random
     >>> random.seed(1234)
-    >>> OmegaConf.register_new_resolver("cached", random.randint)
     >>> OmegaConf.register_new_resolver(
-    ...              "uncached", random.randint, use_cache=False)
-    >>> c = OmegaConf.create({"cached": "${cached:0,10000}",
-    ...                       "uncached": "${uncached:0,10000}"})
+    ...    "cached", random.randint, use_cache=True
+    ... )
+    >>> OmegaConf.register_new_resolver("uncached", random.randint)
+    >>> c = OmegaConf.create(
+    ...     {
+    ...         "cached_1": "${cached:0,10000}",
+    ...         "cached_2": "${cached:0, 10000}",
+    ...         "uncached": "${uncached:0,10000}",
+    ...         "cached_3": "${cached:0,${uncached}}",
```

Comment:
This looks like a better order, first demonstrate the problem that ocurres, then the solution, then the limitations.
```suggestion
    ...         "uncached": "${uncached:0,10000}",
    ...         "cached_1": "${cached:0,10000}",
    ...         "cached_2": "${cached:0, 10000}",
    ...         "cached_3": "${cached:0,${uncached}}",
```

Reply:
Done in a84e17d
(also rebased on top of master)